integration/test/performancetest/performance_test.go

```diff
@@ -1,59 +1,246 @@
 
-	test.CopyFile(configPath, configOutputPath)
-
-	test.StartAgent(configOutputPath, true)
-
-	agentRunDuration := agentRuntimeMinutes * time.Minute
-	//let agent run before collecting performance metrics on it
-	time.Sleep(agentRunDuration)
-	log.Printf("Agent has been running for : %s\n", (agentRunDuration).String())
-	test.StopAgent()
-
-	//collect data
-	data, err := GetPerformanceMetrics(instanceId, agentRuntimeMinutes, agentContext)
-	//@TODO check if metrics are zero remove them and make sure there are non-zero metrics existing
+	configFilePath, logStreams, err := GenerateConfig(logNum)
 	if err != nil {
-		log.Println("Error: ", err)
 		t.Fatalf("Error: %v", err)
 	}
 
-	//------Placeholder to put data into database------//
-	//useless code so data get used and compiler isn't mad
-	if data == nil {
-		t.Fatalf("No data")
+	//defer deleting log group and streams
+	for _, logStream := range logStreams {
+		defer test.DeleteLogGroupAndStream(instanceId, logStream)
+	}
+
+	log.Printf("config generated at %s\n", configFilePath)
+	defer os.Remove(configFilePath)
+
+	tpsVals := []int {
+		10,
+		100,
+		1000,
 	}
-	//data base 
+	
+
+	//data base
 	dynamoDB := InitializeTransmitterAPI(DynamoDBDataBase) //add cwa version here
-	if dynamoDB == nil{
+	if dynamoDB == nil {
 		t.Fatalf("Error: generating dynamo table")
 	}
-	_, err = dynamoDB.SendItem(data)
-	if err !=nil{
-		t.Fatalf("Error: couldnt upload metric data to table")
+
+	//run tests
+	for _, tps := range tpsVals {
+		t.Run(fmt.Sprintf("TPS run: %d", tps), func(t *testing.T) {
+			test.CopyFile(configFilePath, configOutputPath)
+
+			test.StartAgent(configOutputPath, true)
+
+			agentRunDuration := agentRuntimeMinutes * time.Minute
+
+			err := StartLogWrite(agentRunDuration, configFilePath, tps)
+			if err != nil {
+				t.Fatalf("Error: %v", err)
+			}
+
+			log.Printf("Agent has been running for : %s\n", (agentRunDuration).String())
+			test.StopAgent()
+
+			//collect data
+			data, err := GetPerformanceMetrics(instanceId, agentRuntimeMinutes, logNum, tps, agentContext, configFilePath)
+			
+			//@TODO check if metrics are zero remove them and make sure there are non-zero metrics existing
+			if err != nil {
+				t.Fatalf("Error: %v", err)
+			}
+
+			if data == nil {
+				t.Fatalf("No data")
+			}
+
+			_, err = dynamoDB.SendItem(data)
+			if err != nil {
+				t.Fatalf("Error: couldn't upload metric data to table, %v", err)
+			}
+		})
+	}
+}
+
+/* GenerateConfig takes the number of logs to be monitored and applies it to a default config (at ./resources/config.json)
+* it writes logs to be monitored of the form /tmp/testNUM.log where NUM is from 1 to number of logs requested to
+* ./resources/configNUM.json where NUM is number of logs
+* DEFAULT CONFIG MUST BE SUPPLIED WITH AT LEAST ONE LOG BEING MONITORED 
+* (log being monitored will be overwritten - it is needed for json structure)
+*/
+func GenerateConfig(logNum int) (string, []string, error) {
```

Comment:
Do you need to return the list of file paths, or can we cut that out and just call `GetLogFilePaths()` when needed?

Reply:
It's a list of log stream names, not file paths